onset_maproom/maproom.py

```diff
@@ -105,74 +105,301 @@ def map_click(click_lat_lng):
     )
     onsetDate_graph.update_traces(
-        mode="markers+lines",
-        hovertemplate='%{y} %{x}',
-        connectgaps=False
+        mode="markers+lines", hovertemplate="%{y} %{x}", connectgaps=False
     )
     onsetDate_graph.update_layout(
-        yaxis=dict(tickformat="%b %d"), 
-        xaxis_title="Year", 
+        yaxis=dict(tickformat="%b %d"),
+        xaxis_title="Year",
         yaxis_title="Onset Date",
-        title= f"Starting dates of {int(search_start_day)} {search_start_month} season {year.min()}-{year.max()} ({round_latLng(lat)}N,{round_latLng(lng)}E)"
+        title=f"Starting dates of {int(search_start_day)} {search_start_month} season {year.min()}-{year.max()} ({round_latLng(lat1)}N,{round_latLng(lng1)}E)",
     )
-    probExceed_graph = px.line(
+    probExceed_onset = px.line(
         data_frame=cumsum,
         x="Days",
         y="probExceed",
     )
-    probExceed_graph.update_traces(
+    probExceed_onset.update_traces(
         mode="markers+lines",
-        hovertemplate= 'Days since Early Start Date: %{x}'+'<br>Probability: %{y:.0%}'
+        hovertemplate="Days since Early Start Date: %{x}" + "<br>Probability: %{y:.0%}",
     )
-    probExceed_graph.update_layout(
+    probExceed_onset.update_layout(
         yaxis=dict(tickformat=".0%"),
         yaxis_title="Probability of Exceeding",
-        xaxis_title=f"Onset Date [days since {search_start_day} {search_start_month}]"
+        xaxis_title=f"Onset Date [days since {search_start_day} {search_start_month}]",
     )
-    return onsetDate_graph, probExceed_graph, None
-    
+    return onsetDate_graph, probExceed_onset, None
+
+
+@APP.callback(
+    Output("cessDate_plot", "figure"),
+    Output("probExceed_cess", "figure"),
+    Output("coord_alert_cess", "children"),
+    Input("map", "click_lat_lng"),
+    Input("start_cess_day", "value"),
+    Input("start_cess_month", "value"),
+    Input("searchDaysCess", "value"),
+    Input("waterBalanceCess", "value"),
+    Input("drySpellCess", "value"),
+)
+def cess_plots(
+    click_lat_lng,
+    start_cess_day,
+    start_cess_month,
+    searchDaysCess,
+    waterBalanceCess,
+    drySpellCess,
+):
+    lat, lng = get_coords(click_lat_lng)
+    try:
+        precip = rr_mrg.precip.sel(X=lng, Y=lat, method="nearest", tolerance=0.04)
+        isnan = np.isnan(precip).sum().sum()
+        if isnan > 0:
+            errorFig = pgo.Figure().add_annotation(
+                x=2,
+                y=2,
+                text="No Data to Display",
+                font=dict(family="sans serif", size=30, color="crimson"),
+                showarrow=False,
+                yshift=10,
+                xshift=60,
+            )
+            alert1 = dbc.Alert(
+                "Cess alert: The dataset at the chosen coordinates is empty (NaN). Please choose a different point.",
+                color="danger",
+                dismissable=True,
+            )
+            return errorFig, errorFig, alert1
```

Comment:
Lines 277-291 are copy-pasted many times, in this function and the onset one. The only thing that changes is the error message. I would like to factor them out into a function, or maybe two functions (one for the figure and one for the alert). Although, do we really need the alert? Couldn't we just put the full error message in the figure and dispense with the alert?

Reply:
Okay. We can get rid of the error message and put it inside the figure, I made them separate so that the user would see it right away instead of only when they are on the correct tab. But it does not really matter.

Reply:
@aaron-kaplan Is this something you are going to do or should I work on doing this?

Reply:
If you understand what I'm asking for, it would be great if you could do it. If not, it's probably easier for me to do it myself than to explain.

Reply:
I could give it a shot when I return to pyCPT Maproom but that will be only after I push my first PR on water balance. So I suggest that whoever starts working on this among the 3 of us alerts the other 2 so we don't duplicate work.

Reply:
My understanding is that you are wanting to create a function outside of the callbacks that does the exception handling/error printing, so you would just need to call the function on the data in each callback instead of copy pasting all of the code?

Reply:
That's the essence of it, yes. It might be a little tricky because "printing" the error in this case doesn't actually mean printing something to the terminal, it means displaying something in a web page.